### PR TITLE
Add screenshot selector. Feature related to https://github.com/alvarc…

### PR DESCRIPTION
**Pull request by nkimadusanka**
…arto/url-to-pdf-api/issues/125 request

Add feature screenshot selector to support requested improvement https://github.com/alvarcarto/url-to-pdf-api/issues/125

**Developer Comment**
- New attribute introduced in screenshot object called "selector".
- "selector" attribute could be html css id or class.
- "selector" validate using regex to only accept css id or class.
- POST and GET method have been dev tested.

Samples

1. Post request Example

`curl --location --request POST 'http://localhost:9000/api/render' \
--header 'Content-Type: application/json' \
--header 'Content-Type: application/javascript' \
--data-raw '{
	"output": "screenshot",
    "url": "https://expressjs.com/en/starter/hello-world.html",
    "screenshot": {
    	"selector": "#license"
    }
}'`

2. Get request example

http://localhost:9000/api/render?output=screenshot&screenshot.selector=%23license&url=https://expressjs.com/en/starter/hello-world.html

**Note**

- .vscode/launch.json file add to the project to make it easy to debug.

### CODE REVIEW

**Comment by tomasc** [comment]
@nkimadusanka thanks a lot for making this PR, very kind of you!

File: src/core/render-core.js [hunk 1]
```diff
@@ -162,8 +162,14 @@ async function render(_opts = {}) {
       if (clipContainsSomething) {
         screenshotOpts.clip = opts.screenshot.clip;
       }
-
-      data = await page.screenshot(screenshotOpts);
+      if(_.isNil(opts.screenshot.selector)) {
```
Comment (kimmobrunfeldt) [inline src/core/render-core.js:165]: Can you add spacing after if statement here?

File: src/core/render-core.js [hunk 2]
```diff
@@ -162,8 +162,14 @@ async function render(_opts = {}) {
...
+        data = await page.screenshot(screenshotOpts);
+      }else {
```
Comment (kimmobrunfeldt) [inline src/core/render-core.js:167]: Same here

**Comment by kimmobrunfeldt** [comment]
Thanks for your contribution! This was merged via #128.

**Pull request merged**